sycl/source/detail/program_manager/program_manager.cpp

```diff
@@ -158,6 +164,24 @@ string_class ProgramManager::getProgramBuildLog(const RT::PiProgram &Program) {
   return Log;
 }
 
+bool ProgramManager::programContainsKernel(const RT::PiProgram Program,
+                                           const string_class &KernelName) {
+  size_t Size;
+  PI_CALL(RT::piProgramGetInfo(Program, CL_PROGRAM_KERNEL_NAMES, 0, nullptr,
+                               &Size));
+  string_class ClResult(Size, ' ');
+  PI_CALL(RT::piProgramGetInfo(Program, CL_PROGRAM_KERNEL_NAMES,
+                               ClResult.size(), &ClResult[0], nullptr));
+  // Get rid of the null terminator
+  ClResult.pop_back();
```

Comment:
What issues does this solve, exactly?

Reply:
Although no longer a part of the code affected by this patch, this removes the null terminator from the last kernel name in the string returned by OpenCL, so that the checks on the following line work as expected for that kernel.